cmds/netcat/netcat.go

```diff
@@ -18,30 +18,57 @@ import (
 
 const usage = "netcat [go-style network address]"
 
-var netType = flag.String("net", "tcp", "What net type to use, e.g. tcp, unix, etc.")
+var (
+	netType = flag.String("net", "tcp", "What net type to use, e.g. tcp, unix, etc.")
+	listen  = flag.Bool("l", false, "Listen for connections.")
+	verbose = flag.Bool("v", false, "Verbose output.")
+)
 
 func init() {
 	util.Usage(usage)
 }
 
 func main() {
-	flag.Parse()
 	var c net.Conn
 	var err error
-	if len(flag.Args()) < 1 {
+	if flag.Parse(); len(flag.Args()) != 1 {
 		flag.Usage()
 		os.Exit(1)
 	}
 
-	if c, err = net.Dial(*netType, flag.Args()[0]); err != nil {
-		log.Fatalf("%v\n", err)
+	addr := flag.Args()[0]
+
+	if *listen {
+		ln, err := net.Listen(*netType, addr)
+		if err != nil {
+			log.Fatalln(err)
+		}
+		if *verbose {
+			fmt.Fprintln(os.Stderr, "Listening on", ln.Addr())
+		}
+
+		c, err = ln.Accept()
+		if err != nil {
+			log.Fatalln(err)
+		}
+	} else {
+		if c, err = net.Dial(*netType, addr); err != nil {
+			log.Fatalln(err)
+		}
+	}
+	if *verbose {
+		fmt.Fprintln(os.Stderr, "Connected to", c.RemoteAddr())
 	}
+
 	go func() {
 		if _, err := io.Copy(c, os.Stdin); err != nil {
-			fmt.Printf("%v", err)
+			fmt.Fprintln(os.Stderr, err)
```

Comment:
I think I'd prefer `fmt.Fprintf(os.Stderr, "%v", err)`, since errors do not necessarily have stringers.

Reply:
Hmm, Golang specs say that `error` is a stinger interface: https://golang.org/ref/spec#Errors
Do you mean that it can be an empty string sometimes?

Reply:
error is not a fmt.Stringer is what I meant to say. I had assumed Fprintln only took fmt.Stringers, but I'm wrong it seems :)